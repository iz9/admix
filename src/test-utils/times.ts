export function times(amount: number, cb: (...args: any) => any) {
  let i = amount
  while (i > 0) {
    !void cb() && --i
  }
}
