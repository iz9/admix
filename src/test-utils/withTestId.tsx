import React from 'react'

interface WithTestID {
  testId: string
}

export function withTestId<P>(Component: React.ComponentType<P>) {
  const WithTestId: React.FC<P & WithTestID> = props => {
    const { testId, ...rest } = props
    return <Component data-testid={testId} {...(rest as P)} />
  }
  return WithTestId
}
