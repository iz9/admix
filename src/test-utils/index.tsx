import React from 'react'
import { render, RenderOptions } from '@testing-library/react'
import { withTestId } from './withTestId'
import { ThemeProvider } from '../ui-lib/theme'
import { hexToRGB } from './hexToRgb'

const AllTheProviders: React.FC = ({ children }) => (
  <ThemeProvider>{children as React.ReactChild}</ThemeProvider>
)

const customRender = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>,
) => render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render, withTestId, hexToRGB }
