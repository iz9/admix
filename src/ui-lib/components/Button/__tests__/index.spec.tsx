import React from 'react'
import { render, screen, cleanup } from 'test-utils'
import { times } from 'test-utils'
import { Button } from '..'

describe('<Button  />', () => {
  let fn: () => void
  beforeEach(() => {
    fn = jest.fn()
  })
  afterEach(() => {
    cleanup()
  })

  it('should match snapshot', () => {
    const loadingIndicator = render(<Button onClick={fn}>Button</Button>)
    expect(loadingIndicator.container.firstChild).toMatchSnapshot()
  })

  it('should call onClick cb', () => {
    const text = 'Button'
    render(<Button onClick={fn}>{text}</Button>)
    const elem = screen.getByText(text)
    elem.click()
    expect(fn).toHaveBeenCalledTimes(1)
    times(10, elem.click.bind(elem))
    expect(fn).toHaveBeenCalledTimes(11)
  })

  it('should not click when disable', () => {
    const text = 'Disable'
    render(
      <Button disabled onClick={fn}>
        {text}
      </Button>,
    )
    const elem = screen.getByText(text)
    elem.click()
    expect(fn).toHaveBeenCalledTimes(0)
    times(10, elem.click.bind(elem))
    expect(fn).toHaveBeenCalledTimes(0)
  })
})
