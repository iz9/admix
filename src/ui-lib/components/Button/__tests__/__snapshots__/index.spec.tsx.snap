// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button  /> should match snapshot 1`] = `
<button
  class="css-55euh5-Button"
>
  Button
</button>
`;
